source/_docs/guides/jenkins.md

```diff
@@ -0,0 +1,310 @@
+---
+title: Automatically Test and Deploy GitHub Changes to Pantheon from an Existing Jenkins Server
+description: Configure an existing Jenkins server to automatically test and deploy code changes to Pantheon when pushed to GitHub.
+tags: [siteintegrations]
+type: guide
+permalink: docs/guides/:basename/
+date: 6/20/2017
+---
+
+[Jenkins](https://jenkins.io) is an open source Continuous Integration (CI) server which can be used to build, test, and deploy code on any Drupal and WordPress website on Pantheon. Unlike hosted services such as [CircleCI](https://circleci.com/), it is a Java application installed and run on a server, and requires regular upkeep and maintenance. However, it is very customizable and can even run non-CI related tasks, such as calling periodic cron jobs.
+
+In this guide, we'll configure your existing Jenkins server to automatically test and deploy Drupal 8 to a Multidev environment on Pantheon following a code push to your GitHub repository. Jenkins will run tests against both the pull request and Pantheon's master branch and display the results.
+
+## Before You Begin
+
+You will need:
+
+- Root access to a server running Jenkins.
+- Install the following applications, both locally and on the Jenkins server. Verify the default Jenkins user has the ability to run them on the Jenkins server from the command line.
+    - [Git](https://git-scm.com/)
+    - [Composer](https://getcomposer.org/)
+    - [Drush](http://docs.drush.org/en/8.x/install-alternative)
+    - [Terminus](https://github.com/pantheon-systems/terminus#installing-with-composer)
+    - [Terminus Build Tools plugin](https://github.com/pantheon-systems/terminus-build-tools-plugin)
+
+    Verify you can run Terminus, Drush and Composer commands as the Jenkins user. Note that Terminus and Drush will need to be accessible from standard `PATH` directories.
+- Be sure the default Jenkins user has a [private key](https://stackoverflow.com/questions/37331571/how-to-setup-ssh-keys-for-jenkins-to-publish-via-ssh) in its `$HOME/.ssh` directory, and a user with the matching public key installed on Pantheon. Clone a Pantheon site to the Jenkins user's home directory to test.
```

Comment:
If the private key has to be the same for a Pantheon user, let's link to our doc on generating SSH keys. Do we want to recommend a machine user for the Pantheon account? (Seen that question a few times specifically in power users so I think it'd be good to address)

Can we move this out of the before you start section and maybe have it be a new section? The section could be "Readying the Jenkins Server" where lines 27 - 37 are covered with a bit more guidance.